lib/cisco_node_utils/span_session.rb

```diff
@@ -0,0 +1,125 @@
+# Insert appropriate license here
+require_relative 'node_util'
+require_relative 'interface'
+
+
+module Cisco
+  # node utils class for SPAN (switchport analyzer) sessions
+  class SpanSession < NodeUtil
+    attr_reader :session_id
+
+    def initialize(session_id)
+      validate_args(session_id)
+    end
+
+    def self.sessions
+      hash = {}
+      all = config_get('span_session', 'all_sessions')
+      return hash if all.nil?
+
+      all.each do |id|
+        id = id.downcase
+        hash[id] = SpanSession.new(id)
```

Comment:
Once you add the `instantiate` argument to initialize above, call this with `instantiate` set to `false`.  This way you create a list of objects to manage each `span_session` object but you don't try and configure the device since they already exist.

`hash[id] = SpanSession.new(id, false)`